src/Libraries/CoreNodes/List.cs

```diff
@@ -235,6 +236,87 @@ public static IDictionary Deconstruct(IList list)
         }
 
         /// <summary>
+        ///     Sort list based on its keys
+        /// </summary>
+        /// <param name="list">list to be sorted</param>
+        /// <param name="keys">list of keys</param>
+        /// <returns name="sorted list">sorted list</returns>
+        /// <returns name="sorted keys">sorted keys</returns>
+        /// <search>sort;key</search>
+        [MultiReturn(new[] { "sorted list", "sorted keys" })]
+        public static IDictionary SortByKey(IList list, IList keys)
+        {
+            if (list == null || keys == null)
+                return null;
+
+            var containsSublists = keys.Cast<object>().Any(key => key is IList || key is ICollection);
+            if (containsSublists)
+            {
+                throw new ArgumentException(Resources.InvalidKeysErrorMessage);
+            }
+
+            if (list.Count != keys.Count)
+                return null;
```

Comment:
There wasn't any special reason why we didn't show warning. The only reason why we didn't do it, because it was design script and in design there is no opportunity to generate warning. At this place I can throw exception with some clear description. 
@Racel , would you like to show warning, if number of list items doesn't equal number of key items? If you do want, can you provide appropriate message?
Thanks.